src/command.c

```diff
@@ -57,14 +58,19 @@ static void itoa_rbin(uint16_t value, char *out) {
 
     for (int8_t i = 0; i < 16; i++) {
         v = (value >> i) & 1;
-        if (dont_ignore_zeros || v) {
-            out[index++] = '0' + v;
-            dont_ignore_zeros = 1;
-        }
+        out[index++] = '0' + v;
     }
-
-    if (!dont_ignore_zeros) out[index++] = '0';
     out[index] = '\0';
+
+    if (!dont_ignore_zeros) {
+        for (int8_t i = 16; i > 0; i--) {
+            if (out[i] == '1') {
+                index = i;
+                break;
+            }
+        }
+        out[index + 1] = '\0';
```

Comment:
if i'm reading correctly, with the value of `0` it will incorrectly output `0000000000000000` and it will place an extra `\0` out of bounds of `out` array.